### ISSUE TITLE
quotes from Genesis are failing?

### ISSUE THREAD

**Issue by JohnRDOrazio** [BibleGet-I-O/endpoint#29]
For whatever strange reason, I just encountered an error from the WordPress plugin, about a request failing for a quote from Genesis.  So I manually tried the same request directly against the endpoint, and to my surprise it is failing miserably:

https://query.bibleget.io/v3/index.php?query=Genesis1,1-10&version=NVBSE

results in `HTTP ERROR 500`

whereas https://query.bibleget.io/index.php?query=Gen1,1&version=NABRE is correctly returning data:

```js
{
"results": [
{
"testament": 1,
"section": 1,
"book": "Genesis",
"chapter": 1,
"versedescr": null,
"verse": "1",
"verseequiv": null,
"verseorigin": null,
"text": "In the beginning, when God created the heavens and the earth—",
"title1": "",
"title2": "",
"title3": "",
"version": "NABRE",
"bookabbrev": "Gn",
"booknum": 0,
"univbooknum": "1",
"originalquery": "Gen1,1"
}
],
"errors": [],
"info": {
"ENDPOINT_VERSION": "2.9",
"detectedNotation": "EUROPEAN"
}
}
```

What could possible be going wrong? Could it have something to do with `booknum` being zero indexed, and the `0` value of Genesis?

**Comment by JohnRDOrazio** [comment]
Turning on error reporting gives this error:

```
<br />
<b>Fatal error</b>:  Uncaught TypeError: QUERY_FORMULATOR::bestGuessBookIdx(): Return value must be of type int, none returned in /var/www/vhosts/bibleget.io/httpdocs/query/v3/includes/QueryFormulator.php:149
Stack trace:
#0 /var/www/vhosts/bibleget.io/httpdocs/query/v3/includes/QueryFormulator.php(154): QUERY_FORMULATOR-&gt;bestGuessBookIdx()
#1 /var/www/vhosts/bibleget.io/httpdocs/query/v3/includes/QueryFormulator.php(240): QUERY_FORMULATOR-&gt;setBookAndVariant()
#2 /var/www/vhosts/bibleget.io/httpdocs/query/v3/includes/BibleGetQuote.php(615): QUERY_FORMULATOR-&gt;FormulateSQLQueries()
#3 /var/www/vhosts/bibleget.io/httpdocs/query/v3/index.php(126): BIBLEGET_QUOTE-&gt;Init()
#4 {main}
  thrown in <b>/var/www/vhosts/bibleget.io/httpdocs/query/v3/includes/QueryFormulator.php</b> on line <b>149</b><br />
```

**Comment by JohnRDOrazio** [comment]
In `bestGuessBookIdx`, `$this->currentVariant` is empty when the book is `Genesis` or `Gen` or `Gn`, whereas it's correctly set to `NABRE` or 'NVBSE` when the book is anything other than `Genesis`. Why is that?

**Comment by JohnRDOrazio** [comment]
That was exactly the problem. I was checking "truthy" values instead of checking for true boolean values:

https://github.com/BibleGet-I-O/endpoint/blob/95846c66e96e4ab60cc01bc1243897f7c8d23cd2/includes/QueryFormulator.php#L138-L149

In case any of `$key1`, `$key2`, or `$key3` had a value of `0`, it would always evaluate to false. Solution is to check `if !== false`.

**Comment by JohnRDOrazio** [comment]
fixed in commit 4297222c7699142f6754af149ec0914291ae0726
